buildSrc/src/main/kotlin/org/cqfn/save/generation/Generation.kt

```diff
@@ -0,0 +1,196 @@
+import com.squareup.kotlinpoet.ParameterSpec
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import com.squareup.kotlinpoet.PropertySpec
+import com.squareup.kotlinpoet.TypeSpec
+import com.squareup.kotlinpoet.TypeName
+
+import java.io.File
+import java.io.BufferedReader
+
+/**
+ * The comment that will be added to the generated sources file.
+ */
+private val autoGenerationComment =
+    """
+    | ---------------------------------------------------------------------
+    | ******* This file was auto generated, please don't modify it. *******
+    | ---------------------------------------------------------------------
+    """.trimMargin()
+
+// Paths, where to store generated files
+val generatedSaveSavePropertiesPath = "save-core/src/commonMain/kotlin/org/cqfn/save/core/config/"
+val generatedOptionsTablePath = "buildSrc/src/main/kotlin/org/cqfn/save/generation/"
+
+/**
+ * This class represents the general form of each key in json file with config options
+ * @property argType Type which will be used in ArgParser in Config.kt
+ * @property kotlinType Type which will be used in kotlin code
+ * @property fullName Full name of option for usage in Save cli
+ * @property shortName Short name of option for usage in Save cli
+ * @property description Option description
+ * @property default default value for option
+ */
+class Option {
+    lateinit var argType: String
+    lateinit var kotlinType: String
+    lateinit var fullName: String
+    lateinit var shortName: String
+    lateinit var description: String
+    lateinit var default: String
+}
+
+fun main() {
+    val configFile = "buildSrc/src/main/kotlin/config-options.json"
+    val gson = Gson()
+    val bufferedReader: BufferedReader = File(configFile).bufferedReader()
+    val jsonString = bufferedReader.use { it.readText() }
+    val jsonObject = gson.fromJson<Map<String, Option>>(jsonString, object : TypeToken<Map<String, Option>>(){}.type)
+    generateSaveProperties(jsonObject)
+    generateReadme(jsonObject)
+}
+
+fun generateSaveProperties(jsonObject: Map<String, Option>) {
+    val builder = FileSpec.builder("org.cqfn.save.core.config", "SaveProperties")
+    builder.addComment(autoGenerationComment)
+    builder.addImport("kotlinx.cli", "ArgParser")
+    builder.addImport("kotlinx.cli", "ArgType")
+    val classBuilder = generateSavePropertiesClass(jsonObject)
+    val mergeFunc = generateMergeConfigFunc(jsonObject)
+    classBuilder.addFunction(mergeFunc.build())
+    builder.addType(classBuilder.build())
+    builder.build().writeTo(System.out)
+    File("$generatedSaveSavePropertiesPath/SaveProperties.kt").writeText(builder.build().toString())
+}
+
+fun generateSavePropertiesClass(jsonObject: Map<String, Option>): TypeSpec.Builder {
+    val classBuilder = TypeSpec.classBuilder("SaveProperties")
+    var properties = ""
+    jsonObject.forEach { properties += ("@property ${it.key} ${it.value.description}\n")}
```

Comment:
This can be combined with `jsonObject.joinToString`

Reply:
Done